luda-adventure/src/app/game/types/game_object/types/positioner/positioner.rs

```diff
@@ -32,14 +32,15 @@ impl Positioner {
     pub fn predict_movement(
         &mut self,
         collider: &Collider,
-        collision_box_list: &[crate::app::game::CollisionBox],
+        target_collision_box_list: &[crate::app::game::CollisionBox],
     ) {
-        let prediction_start_time = self.get_predicted_movement_end_time();
-        let prediction_start_position = self.get_position(prediction_start_time);
+        let last_prediction = self.get_last_prediction();
+        let prediction_start_time = last_prediction.end_time;
+        let prediction_start_position = last_prediction.end_position;
         let character_collision_box = collider.get_collision_box(prediction_start_position);
+        let directed_movement = &self.movement_plan.directed_movement;
```

Comment:
What's the meaning `directed_movement` for ?

Reply:
Ideal movement to be executed without any collision.
User directed movement.

Reply:
It makes some confusing because it looks like about direction (--> , <--)

Reply:
requested_movement?